x-pack/plugins/maps/public/classes/styles/vector/properties/components/breaked_legend.js

```diff
@@ -0,0 +1,108 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import React from 'react';
+import _ from 'lodash';
+import { Category } from '../../components/legend/category';
+import { EuiFlexGroup, EuiFlexItem, EuiText, EuiTextColor, EuiToolTip } from '@elastic/eui';
+import { getOtherCategoryLabel } from '../../style_util';
+const EMPTY_VALUE = '';
+
+export class BreakedLegend extends React.Component {
+  state = {
+    label: EMPTY_VALUE,
+  };
+
+  componentDidMount() {
+    this._isMounted = true;
+    this._loadParams();
+  }
+
+  componentDidUpdate() {
+    this._loadParams();
+  }
+
+  componentWillUnmount() {
+    this._isMounted = false;
+  }
+
+  async _loadParams() {
+    const label = await this.props.style.getField().getLabel();
+    const newState = { label };
+    if (this._isMounted && !_.isEqual(this.state, newState)) {
+      this.setState(newState);
+    }
+  }
+
+  render() {
+    if (this.state.label === EMPTY_VALUE) {
+      return null;
+    }
+
+    const categories = [];
+
+    this.props.stops.map(({ stop, color, style }) => {
```

Comment:
Not sure if this will work when this file gets converted to typescript. Instead of passing in stops which are two different types for icons and colors, pass in array of Category elements so that way the different logic for handling colors and icons and stay in DynamicColorProperty and DynamicIconProperpty

Reply:
Let's do this in another PR, when we'd actually do this TS conversion. There's already enough refactoring going on. In the current codebase icon and color palettes do have a different shape, but this imho this is not the PR to align those (and such a change would bleed into some of the existing utility code, further compounding changes here).